identity/did.go

```diff
@@ -29,22 +28,31 @@ const (
 	// BootstrappedDIDService stores the id of the service
 	BootstrappedDIDService string = "BootstrappedDIDService"
 
-	// CentIDLength is the length in bytes of the DID
-	CentIDLength = 6
-
-	// KeyPurposeP2P represents a key used for p2p txns
-	KeyPurposeP2P = 1
-
-	// KeyPurposeSigning represents a key used for signing
-	KeyPurposeSigning = 2
-
-	// KeyPurposeEthMsgAuth represents a key used for ethereum txns
-	KeyPurposeEthMsgAuth = 3
-
 	// KeyTypeECDSA has the value one in the ERC725 identity contract
 	KeyTypeECDSA = 1
 )
 
+// KeyPurposeP2PDiscovery is calculated out of Hex(sha256("CENTRIFUGE@P2P_DISCOVERY"))
+func KeyPurposeP2PDiscovery() *Purpose {
+	hashed := "88dbd1f0b244e515ab5aee93b5dee6a2d8e326576a583822635a27e52e5b591e"
```

Comment:
lets create a constant and use it. I'd actually prefer calculating hash instead of having a hash. What do you think ?

Reply:
better yet, we can have these purposes as global variables that are calculated through init function instead of creating every time through the function call. Better for memory allocation